### PR TITLE
Generated terrain

### PR DESCRIPTION
**Pull request by jorisquist**
Adds support for automatically generated terrain. By default it's not enabled since it is a bit heavier to simulate.

### CODE REVIEW

File: lobster_simulator/common/Terrain.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import math
+
+import noise
```
Comment (JonathanBrouwer) [inline lobster_simulator/common/Terrain.py:3]: noise is not in the requirements

File: lobster_simulator/common/Terrain.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+class Terrain:
+    chunks = dict()
+
+    def __init__(self, depth=100):
+        self.current_chunk = (0, 0)
+
+        # TODO: Changing the chunk size should not impact the shape of the generated world, however for some reason
+        #  it does. If you think you can fix it, go ahead.
+        self.chunk_size = 2**7 - 1
```
Comment (JonathanBrouwer) [inline lobster_simulator/common/Terrain.py:19]: Why is the chunk size 2^7 - 1, rather than just 2^7?

File: lobster_simulator/common/Terrain.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+from lobster_simulator.tools.PybulletAPI import PybulletAPI
+
+
+class Terrain:
+    chunks = dict()
+
+    def __init__(self, depth=100):
+        self.current_chunk = (0, 0)
+
+        # TODO: Changing the chunk size should not impact the shape of the generated world, however for some reason
```
Comment (JonathanBrouwer) [inline lobster_simulator/common/Terrain.py:17]: I switched the terrain generation to sin, and changing the chunk size does not appear to have an effect on the shape of the generated world. Could you clarify what is wrong?

File: lobster_simulator/common/Terrain.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                               persistence=persistence,
+                               lacunarity=lacunarity,
+                               repeatx=1024,
+                               repeaty=1024,
+                               base=seed)
+        height *= 200
+
+        return height
+
+    def get_height_field_sin(self, chunk_x, chunk_y):
```
Comment (JonathanBrouwer) [inline lobster_simulator/common/Terrain.py:50]: Sin is unused, if you intended it to be an alternative for perlin generation, maybe create an abstract class for the type of generation?

File: control/HighLevelController.py [hunk 5]
```diff
@@ -112,14 +112,14 @@ def update(self, position: Vec3, orientation: Quaternion, velocity: Vec3, angula
         # self.acceleration_pids[Z].set_target(self.velocity_pids[Z].output)
 
         acceleration = (self.previous_velocity - local_frame_velocity) / dt
-        print(acceleration)
```
Comment (JonathanBrouwer) [inline control/HighLevelController.py:115]: There are quite a few leftover print statements which spam the console, you removed this one but maybe also remove the others?

File: lobster_simulator/common/Terrain.py (hunk 2 above)
Comment (jorisquist) [reply lobster_simulator/common/Terrain.py:19]: That was done to make sure that if you also pick 2**7 points per chunk than you the points are exactly 1 meter apart.

File: lobster_simulator/common/Terrain.py (hunk 3 above)
Comment (jorisquist) [reply lobster_simulator/common/Terrain.py:17]: Yes it only happens when using perlin noise for the height map. What happens is that, if you use smaller chunk sizes than the whole world seems to shrink in the x and y direction. It is probably an error somewhere with a scaling factor, but I couldn't find it and I didn't want to spend to much time on it.

File: lobster_simulator/common/Terrain.py (hunk 4 above)
Comment (jorisquist) [reply lobster_simulator/common/Terrain.py:50]: Yeah, it was mainly used for testing, but I now gave the Terrain constructor a parameter which is a height function

File: control/HighLevelController.py (hunk 5 above)
Comment (jorisquist) [reply control/HighLevelController.py:115]: I think I have them all.

File: lobster_simulator/common/Terrain.py (hunk 1 above)
Comment (jorisquist) [reply lobster_simulator/common/Terrain.py:4]: I added it.

File: lobster_simulator/common/Terrain.py (hunk 2 above)
Comment (JonathanBrouwer) [reply lobster_simulator/common/Terrain.py:20]: I don't think that logic works, unless you count points on the edge of chunks as part of both chunks

File: lobster_simulator/common/Terrain.py (hunk 3 above)
Comment (gijskoning) [reply lobster_simulator/common/Terrain.py:18]: Could it be something to do with the float -> int casting?

**Pull request merged**